src/uct/ib/base/ib_iface.c

```diff
@@ -743,6 +723,39 @@ int uct_ib_iface_is_reachable(const uct_iface_h tl_iface,
     }
 }
 
+int uct_ib_iface_is_reachable_v2(const uct_iface_h tl_iface,
+                                 const uct_iface_is_reachable_params_t *params)
+{
+    uct_ib_iface_t *iface           = ucs_derived_of(tl_iface, uct_ib_iface_t);
+    const uct_ib_address_t *ib_addr = (const void*)params->device_addr;
+    uct_iface_reachability_scope_t scope;
+
+    if (!uct_ib_iface_addr_is_reachable(iface, params->device_addr,
+                                        params->iface_addr)) {
+        return 0;
+    }
+
+    scope = UCS_PARAM_VALUE(UCT_IFACE_IS_REACHABLE_FIELD, params, scope, SCOPE,
+                            UCT_IFACE_REACHABILITY_SCOPE_NETWORK);
+
+    return (scope == UCT_IFACE_REACHABILITY_SCOPE_NETWORK) ||
+           uct_ib_iface_is_same_device(iface, ib_addr);
+}
+
+int uct_ib_iface_is_reachable(const uct_iface_h tl_iface,
+                              const uct_device_addr_t *dev_addr,
+                              const uct_iface_addr_t *iface_addr)
+{
+    uct_iface_is_reachable_params_t params = {
+        .field_mask  = UCT_IFACE_IS_REACHABLE_FIELD_DEVICE_ADDR |
+                       UCT_IFACE_IS_REACHABLE_FIELD_IFACE_ADDR,
+        .device_addr = dev_addr,
+        .iface_addr  = iface_addr
+    };
+
+    return uct_ib_iface_is_reachable_v2(tl_iface, &params);
```

Comment:
@brminich, following your comment, I think you are right in the sense that this line should rather be `uct_iface_is_reachable_v2()`/`->iface_is_reachable_v2()`, to take the dc or rc specific v2.

Reply:
fixed